tests/selectedtests/task_mappings/test_task_mappings_cli.py

```diff
@@ -19,10 +19,18 @@ def m_ns(relative_name):
 
 
 class TestCli:
+    @patch(m_ns("init_repo"))
```

Comment:
I'm confused here. This cli test is an e2e test using mocked evergreen.py responses - so the point is that given certain responses from evergreen, we should, when running the full cli command, get certain task mappings back. In that case, the stuff we don't care about is the actual getting of repos in the other mappings.py file which is what is being mocked here. The other mocks are enabling us to give fake evergreen data and to get the "change files" we need to make mappings.

I can drag the actual initialization of the repo(s) out of this code and make the cli do it and pass the repos in as parameters. I'm not sure that fits the comment you linked as that seems to be adding more functionality to the cli command code.

Or is the ask to drag both of those functions into a single wrapper function? Or is it to make this test more like the test-mappings-cli tests and mock out the ```create_task_mappings``` function entirely and just make them test that the cli command parses and passes info?

Reply:
If this is meant to be an e2e test, then we want minimize what is being mocked and ideally only mock external parts of the system. So I would not expect `init_repo`, `get_evg_project`, or `_get_filtered_files` to be mocked, since they is not an external part of the system. Instead I would expect `Repo` and `RetryingEvergreenApi` to be mocked since they are the external parts of the system.

Reply:
Sure, e2e was probably the wrong word choice. It's meant to be a test to make sure that, given certain evergreen data, the correct task mappings are created and returned to the cli in the manner we expect. All the ```repo``` stuff is superfluous to that so that's mocked out, which means we need to mock the ```changed_files``` we find, and the ```RetryingEvergreenApi``` is mocked to ensure that we get the evergreen data we need.

@lydiastepanek As discussed, I've created a helper function called ```_get_evg_project_and_init_repo``` that wraps those two functions from earlier and also added some additional tests to the cli.

@dbradf As discussed, I've removed my mock of ```get_evg_project``` and instead added additional information to the ```evg_api``` mock.